pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -134,6 +147,26 @@ func createKnativeService(componentMeta metav1.ObjectMeta,
 	return service
 }
 
+func addAgentAnnotations(component constants.InferenceServiceComponent, isvc *v1beta1.InferenceService, annotations map[string]string) bool {
+	if component == constants.Predictor && isvc.Spec.Predictor.GetImplementation().GetStorageUri() == nil {
+		annotations[constants.AgentInternalAnnotationKey] = "true"
+		shardStrategy := memory.MemoryStrategy{}
+		for _, id := range shardStrategy.GetShard(isvc) {
+			modelConfig, err := modelconfig.CreateEmptyModelConfig(isvc, id)
+			if err == nil {
+				annotations[constants.AgentModelConfigAnnotationKey] = modelConfig.Name
+				//TODO figure out the proper way to inject s3 endpoint from s3 identity
+				annotations[constants.AgentS3endpointAnnotationKey] = "gs://kfserving-samples"
```

Comment:
Why is this needed? The s3 endpoint should be injected in the container in the agent sidecar.

Reply:
Oops forgot to delete this line, will remove.